src/iotHubC2DMessageExplorer.ts

```diff
@@ -51,13 +51,13 @@ export class IotHubC2DMessageExplorer extends BaseExplorer {
         this._deviceClient.open(this.connectCallback(deviceConnectionString));
     }
 
-    public stopMonitorC2DMessage(): void {
+    public stopMonitorC2DMessage() {
         TelemetryClient.sendEvent(Constants.IoTHubAIStopMonitorC2DEvent);
         this._outputChannel.show();
-        if (this._deviceClient) {
+        if (this._isMonitoring) {
             this.outputLine(Constants.IoTHubC2DMessageMonitorLabel, "C2D monitoring stopped.");
-            this._deviceClient.close(() => { return; });
-            this._deviceClient = null;
+            this._monitorStatusBarItem.hide();

```

Comment:
hide will be called in updateMonitorStatus right?

Reply:
Yes, since `this._deviceClient.close()` may take one or more seconds, I explicitly call this to hide the button earlier.

Reply:
Better to have one place to do the stuff. Besides, the user could click the stop button, and then ctrl+c to cancel and the "C2D monitoring stopped" message will show several times. Since actually the monitor has not been stopped.

Reply:
The motivation of calling `this._monitorStatusBarItem.hide()` in advance is to avoid user click stop button several times.